core/cap-0063.md

```diff
@@ -0,0 +1,447 @@
+it must provide the `evictedSorobanEntries` vector containing the index of each evicted key based on the ordering of the
+readWrite footprint. Because restores are a write operation, no evicted key should be in the readOnly footprint. This vector
+should contain no classic key indexes, as classic keys are always considered disk reads.
+
+#### Changes to `InvokeHostFunctionOp`
+
+##### Automatic Entry Restoration
+
+Whenever `InvokeHostFunctionOp` is applied, any archived state is
+automatically restored prior to host function invocation. This restored state is then accessible to the host function.
+
+Restored state is still subject to the same minimum rent and write fees that exist currently.
+
+If an archived Soroban entry has not yet been evicted and still exists in the live BucketList, it is metered and charged based
+on in-memory read limits and fees. If an archived Soroban entry has been evicted, it is metered and charged based
+on on-disk limits and fees. All evicted keys must be declared in the `evictedSorobanEntries` vector via `SorobanResourcesExtV0`.
+If no evicted entries are present in the footprint, `SorobanResourcesExtV0` may be omitted.
+
+##### Incorrect Entries in `evictedSorobanEntries` Vector
+
+If a key is not marked as evicted, but the entry is actually evicted, then the TX fails. If a Soroban key is marked as evicted when
+it is not, the transaction does not fail. The entry is treated as if it is evicted from a resource and fee perspective.
+This is appropriate, as on disk reads are more expensive than in-memory. So long as the TX pays the disk based fees,
+there is no issue with actually loading an in-memory entry.
+
+If a classic key is marked as evicted, the TX fails, as the footprint is malformed.
+
+##### Failure Codes
+
+If an evicted entry is included in the TX's footprint but is not specified via the `evictedSorobanEntries` vector, the
+tx will fail with the `INVOKE_HOST_FUNCTION_ENTRY_ARCHIVED` code. If there are insufficient resources for entry restoration,
+the tx will fail with the `INVOKE_HOST_FUNCTION_RESOURCE_LIMIT_EXCEEDED` code.
+
+If a classic entry is marked as evicted, the TX fails with `INVOKE_HOST_FUNCTION_MALFORMED` during apply time.
+
+##### Meta
+
+Whenever an entry is restored, it will be emitted as a `LedgerEntryChangeType` of type `LEDGER_ENTRY_RESTORE`.
+This will include the complete LedgerEntry of the restored entry. It does not matter if the entry has been
+evicted or not, the meta produced will be the same.
+
+If a restored entry is modified during host function invocation, another `LedgerEntryChangeType` will be
+emitted corresponding to the entry change in addition to the `LEDGER_ENTRY_RESTORE` restore event.
+This is similar to the current meta schema, where the starting value of a LedgerEntry is emitted as
+`LEDGER_ENTRY_STATE` followed by the change. This is similar, except that `LEDGER_ENTRY_STATE` is replaced
+by `LEDGER_ENTRY_RESTORE`.
+
+#### Deprecation of RestoreFootprintOp(?)
```

Comment:
Worth asking the ecosytem on deprecating this Op. Only thing I can think of is that if admin doesn't want the users to incur the restoration fees and keeps expired/archived entries restored on behalf of the users.